awaygoApp/decorators.py

```
@@ -0,0 +1,29 @@
+from django.http import HttpResponse
+from django.shortcuts import redirect
+
+def unauthenticated_user(view_func):
+    def wrapper_func(request, *args, **kwargs):
+        if request.user.is_superuser:
+            return redirect('admin/')
+        elif request.user.is_authenticated:
+            return redirect('restaurant')
```

Since this is just an API, usually it's better to throw a PermissionDenied error, which would make django, if all is configured well, to return an http response with the status code 403. Have a glance in http response codes and memorize them a bit, at least 200 500 403 404.

---

In all of the instances where you are returning a Response with a custom response code, you could use django's PermissionDenied and NotFound exceptions, and additional ones, which django is supposed to handle and return the appropriate code automatically.